### PR TITLE
Adds listening post access to sleeper agents

### PR DESCRIPTION
**Pull request by cheekybrdy**
[BALANCE] [FEATURE]
## About the PR 
This PR is a remake of #14992. It adds handscanners which a sleeper can use to enter listening post to every map.

## Why's this needed? 
It gives sleepers more options to work with without the full power of a uplink. This makes them more then a late hard mode traitor

## Changelog 

```changelog
(u)Cheekybrdy
(*)Sleeper agents can now use a handscanner at the listening post to get into it.
```

### CODE REVIEW

**Comment by cheekybrdy** [comment]
Discussion thread from the original PR: https://forum.ss13.co/showthread.php?tid=21546

**Comment by deathrobotpunch** [comment]
AAAAAAAAAAAA RUNTIMEEEEESSSSSSS

File: code/obj/machinery/sleeperscanner.dm [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	icon = 'icons/obj/decoration.dmi'
+	icon_state = "handscanner"
+	desc = "A device embedded into the wall, it has a image of a handprint on it."
+
+/obj/machinery/sleeperscanner/attackby(obj/item/W, mob/user as mob)
+	if(istype(W, /obj/item/device/detective_scanner))
+		return
+	if (user.mind?.get_antagonist(ROLE_SLEEPER_AGENT))
+		user.visible_message("<span class='notice'>The [src] accepts the biometrics of the user and beeps, granting you access.</span>")
+		for (var/obj/machinery/door/airlock/pyro/reinforced/syndicate/M in by_type[/obj/machinery/door])
```
Comment (Mister-Moriarty) [inline code/obj/machinery/sleeperscanner.dm:12]: The type `/obj/machinery/door/airlock/pyro/reinforced/syndicate` is used in several prefabs and in Z2; listening post doors will require to be their own subtype if this is to work.
I would personally recommend refactoring sleeper hand scanners into a subtype of `/obj/machinery/door_control`, and utilising `id` to open the listening post doors.

**Review by Mister-Moriarty — CHANGES_REQUESTED** [review]

**Comment by cheekybrdy** [comment]
!merge_upstream

**Comment by TheTyrant25** [comment]
might want to start over on a fresh branch at this point

**Pull request closed without merging**